src/trust-network/trust-network.service.ts

```diff
@@ -21,25 +21,47 @@ export class TrustNetworkService {
     const { id, sender, party, associationType } = transaction;
     
     if (!party) {
-      this.logger.debug(`trustNetwork: transaction ${id} didn't have a party address, skipped indexing`);
+      this.logger.debug(`trust-network: transaction ${id} didn't have a party address, skipped indexing`);
       return;
     }
 
     if (!associationType) {
-      this.logger.debug(`trustNetwork: transaction ${id} didn't have an association type, skipped indexing`);
+      this.logger.debug(`trust-network: transaction ${id} didn't have an association type, skipped indexing`);
       return;
     }
 
     const senderRoles = await this.getRolesFor(sender);
 
     const savedRoles: Role[] = [];
 
-    senderRoles.issues_roles.forEach(eachRole => {
+    senderRoles.issues_roles.forEach(async eachRole => {
       if (eachRole.type === associationType && !savedRoles.includes(eachRole)) {
         savedRoles.push(eachRole);
-        this.storage.saveRoleAssociation(party, sender, eachRole);
+        await this.storage.saveRoleAssociation(party, sender, eachRole);
       }
     });
+
+    if (savedRoles.length === 0) {
+      return;
+    }
+
+    const hasSponsoredRole = savedRoles.reduce((previous, current, index) => {
+      const each = savedRoles[index];
+      const configRoles = this.config.getRoles();
+
+      return !!configRoles[each.role]?.sponsored;
+    }, false);
+
+    if (!hasSponsoredRole) {
+      return;
+    }
+
+    try {
+      this.logger.debug(`trust-network: party is being given a sponsored role, sending a transaction to the node`);
+      await this.node.signSponsorTransaction(sender, party);
+    } catch(error) {
+      this.logger.error(`trust-network: error sending a transaction to the node: "${error}"`);
+    }
```

Comment:
Move this to a (private) method. The function is too big.

Reply:
Done